### DISCUSSION TITLE
Can you clarify the PostCSS pipeline?

### DISCUSSION THREAD

**Discussion by hyperknot** [vercel/turborepo#3217] [Help]
I'm trying to debug why the default create-next-app doesn't work with the default PostCSS config ([issue](https://github.com/vercel/next.js/issues/44286)), as written in the Next docs. Can you clarify what is happening there? I guess since Turbo was introduced, that docs wasn't updated? There is no mention about Turbo, SWC or swc_css at all.

What I've found so far is that:
- There is no autoprefixer installed anywhere in node_modules in a Next.js install.
- Yet, it's still being applied because now autoprefixer is compiled into swc via swc_css? I'm just guessing here.
- What is exactly included in swc? autoprefixer + postcss-flexbugs-fixes + postcss-preset-env are all compiled into swc?

Because the moment I specify anything in postcss.config.js, those need to be installed manually.

**Comment by EarlGeorge** [comment]
yes but how it's related to the turbo pack or even to swc ?!

**Reply by hyperknot** [reply]
autoprefixer + postcss-flexbugs-fixes + postcss-preset-env is not included anywhere in the node_modules of Next.jx, yet it works out of the box. It must be compiled into swc I guess, hence I'm asking it here.

I don't understand perfectly the relation between Turbopack and SWC, hence I asked it here.

**Reply by EarlGeorge** [reply]
both bundle what you throw inside. let's ask the most relevant lad, hope he will have some free time to answer your questions @kdy1
